odoo/tools/sql.py

```diff
@@ -313,25 +315,34 @@ def reverse_order(order):
     return ', '.join(items)
 
 
-def increment_field_skiplock(record, field):
+def increment_fields_skiplock(records, *fields):
     """
-        Increment 'friendly' the [field] of the current [record](s)
+        Increment 'friendly' the given `fields` of the current `records`.
         If record is locked, we just skip the update.
         It doesn't invalidate the cache since the update is not critical.
 
-        :rtype: bool - if field has been incremented or not
+        :param records: recordset to update
+        :param fields: integer fields to increment
+        :returns: whether the specified fields were incremented on any record.
+        :rtype: bool
     """
-    if not record:
+    if not records:
         return False
 
-    assert record._fields[field].type == 'integer'
+    for field in fields:
+        assert records._fields[field].type == 'integer'
 
-    cr = record._cr
-    query = """
-        UPDATE {table} SET {field} = {field} + 1 WHERE id IN (
-            SELECT id from {table} WHERE id in %(ids)s FOR UPDATE SKIP LOCKED
-        ) RETURNING id
-    """.format(table=record._table, field=field)
-    cr.execute(query, {'ids': tuple(record.ids)})
+    field_identifiers = tuple(map(Identifier, fields))
 
-    return bool(cr.fetchone())
+    query = SQL("""
+        UPDATE {table}
+           SET ({fields}) = ROW({fields_plus_one})
```

Comment:
FWIW `ROW` doesn't seem necessary, according to the docs:
```sql
UPDATE weather SET (temp_lo, temp_hi, prcp) = (temp_lo+1, temp_lo+15, DEFAULT)
  WHERE city = 'San Francisco' AND date = '2003-07-03';
```
And not sure this method is even necessary? Why not e.g.
```python
sets=SQL(', ').join(map(SQL('{0} = {0} + 1').format, field_identifiers))
```

Reply:
> FWIW `ROW` doesn't seem necessary, according to the docs:
> 
> ```sql
> UPDATE weather SET (temp_lo, temp_hi, prcp) = (temp_lo+1, temp_lo+15, DEFAULT)
>   WHERE city = 'San Francisco' AND date = '2003-07-03';
> ```

This doesn't work when there is only one column.
E.g., `set ("column_a") = (3)` would raise `[0A000] ERROR: source for a multiple-column UPDATE item must be a sub-SELECT or ROW() expression`

> And not sure this method is even necessary? Why not e.g.
> 
> ```python
> SQL(', ').join(map(
>     SQL('{0} = {0} + 1').format,
>     field_identifiers
> ))
> ```
> 
> for the entire `set` contents?

Indeed, it removes that precaution and we can even avoid creating the `fields_identifier` tuple as well.

Reply:
> > FWIW `ROW` doesn't seem necessary, according to the docs:
> > ```sql
> > UPDATE weather SET (temp_lo, temp_hi, prcp) = (temp_lo+1, temp_lo+15, DEFAULT)
> >   WHERE city = 'San Francisco' AND date = '2003-07-03';
> > ```
> 
> This doesn't work when there is only one column. E.g., `set ("column_a") = (3)` would raise `[0A000] ERROR: source for a multiple-column UPDATE item must be a sub-SELECT or ROW() expression`

Ah good call. In that case I'd have wanted a comment to indicate that if we'd kept it.

> Indeed, it removes that precaution and we can even avoid creating the `fields_identifier` tuple as well.

Nice :+1: